app/src/main/java/com/example/tasky/feature_agenda/presentation/event_detail_screen/EventDetailViewModel.kt

```diff
@@ -0,0 +1,194 @@
+): ViewModel() {
+
+    private val _state = MutableStateFlow(EventDetailState())
+    val state = _state.asStateFlow()
+
+    private val photoList = mutableStateListOf<Photo>()
+
+    init {
+        val eventId = savedStateHandle.get<String>(ArgumentTypeEnum.ITEM_ID.name)
+        val editMode = savedStateHandle.get<String>(ArgumentTypeEnum.EDIT_MODE.name)
+
+        eventId?.let {
+            val editable = editMode != null
+            getSelectedEvent(it, editable)
+        }
+    }
+
+    fun onEvent(event: EventDetailOnClick) {
+        when(event) {
+            EventDetailOnClick.SaveEvent -> {
+                //saveEvent()
+            }
+            is EventDetailOnClick.SelectFilterOption -> {
+                _state.update {
+                    it.copy(selectedChip = event.selectedChip)
+                }
+            }
+            EventDetailOnClick.ToggleAddingAttendeeDialog -> {
+                _state.update {
+                    it.copy(addingAttendees = !it.addingAttendees)
+                }
+            }
+            EventDetailOnClick.ToggleAddingPhotos -> {
+                _state.update {
+                    it.copy(addingPhotos = !it.addingPhotos)
+                }
+            }
+            EventDetailOnClick.ToggleEditMode -> {
+                _state.update {
+                    it.copy(editMode = !it.editMode)
+                }
+            }
+            EventDetailOnClick.ToggleReminderMenu -> {
+                _state.update {
+                    it.copy(isReminderMenuVisible = !it.isReminderMenuVisible)
+                }
+            }
+
+            is EventDetailOnClick.DescriptionChanged -> {
+                _state.update {
+                    it.copy(eventDescription = event.description)
+                }
+            }
+            is EventDetailOnClick.TitleChanged -> {
+                _state.update {
+                    it.copy(eventTitle = event.title)
+                }
+            }
+
+            is EventDetailOnClick.AddPhoto -> {
+                if(event.photoUri == null) {
+                    _state.update {
+                        it.copy(addingPhotos = !it.addingPhotos)
+                    }
+                } else {
+                    photoList.add(
+                        Photo(
+                            key = UUID.randomUUID().toString(),
+                            url = event.photoUri.toString()
+                        )
+                    )
+                    _state.update {
+                        it.copy(photoList = photoList)
+                    }
+                }
+            }
+
+            is EventDetailOnClick.DeletePhoto -> {
+                val photoToRemove = photoList.find { it.key == event.photoKey }
+
+                photoToRemove?.let { photo ->
+                    photoList.remove(photo)
+                    _state.update {
+                        it.copy(
+                            photoList = photoList,
```

Comment:
Your `photoList` is already a Compose state list. That will recompose your UI when a new item gets added - you therefore don't need to include this list in your state class